entrypoint.sh

```diff
@@ -92,6 +98,24 @@ initial.load.create.first=true
 EOL
 fi
 
+# basic auth setup!!
+sed -i "s/<\/web-app>/<security-constraint><web-resource-collection><url-pattern>\/sync\/*<\/url-pattern><\/web-resource-collection><auth-constraint><role-name>user<\/role-name><\/auth-constraint><\/security-constraint><login-config><auth-method>BASIC<\/auth-method><realm-name>default<\/realm-name><\/login-config><\/web-app>/g" ./web/WEB-INF/web.xml
```

Comment:
I don't think you need the `g` modifier. It also might be cleaner to use a character besides forward-slash. e.g.
```bash
sed -i "s|</web-app>|<security-constraint><web-resource-collection><url-pattern>/sync/*</url-pattern></web-resource-collection><auth-constraint><role-name>user</role-name></auth-constraint></security-constraint><login-config><auth-method>BASIC</auth-method><realm-name>default</realm-name></login-config></web-app>|" ./web/WEB-INF/web.xml
```